src/components/builder.ts

```diff
@@ -160,9 +160,30 @@ export class Builder {
         this.extension.logger.displayStatus('sync~spin', 'statusBar.foreground')
         this.extension.logger.addLogMessage(`Build root file ${rootFile}`)
         try {
-            this.extension.buildInfo.buildStarted()
-            pdfjsLib.getDocument(this.extension.manager.tex2pdf(rootFile, true)).promise.then((doc: any) => {
-                this.extension.buildInfo.setPageTotal(doc.numPages)
+            const configuration = vscode.workspace.getConfiguration('latex-workshop')
+            let location: vscode.ProgressLocation
+            let title: string
+            if ((configuration.get('progress.location') as string) === 'Status Bar') {
+                location = vscode.ProgressLocation.Window
```

Comment:
Two issues with using the status-bar progress report:
 1. Adds a redundant spinner (too many icons)
![image](https://user-images.githubusercontent.com/20903656/66841440-4995c700-ef9c-11e9-9b6a-2192cd6d1b9e.png)
 2. This spinner resets every time a new page is pushed out, which makes it look glitchy

Reply:
I can remove the spinner from the logger status bar item, I think we're stuck with the spinner on the progress bar and I don't think I can do anything about it resetting.

Reply:
Erm, surely if it works as a status just leave it as a status?

Reply:
That is probably best, the more I think about it and work on it. The native progress API only offers improvement for notifications, for the status bar the API only allows us to display a message (which we can already do).